opentelemetry-sdk/src/opentelemetry/sdk/_metrics/_internal/metric_reader_storage.py

```diff
@@ -116,13 +128,89 @@ def collect(self) -> Iterable[Metric]:
+                view_instrument_matches,
+            ) in self._instrument_view_instrument_matches.items():
+                aggregation_temporality = self._instrument_class_temporality[
+                    instrument.__class__
+                ]
+
+                metrics: List[Metric] = []
+
                 for view_instrument_match in view_instrument_matches:
-                    metrics.extend(view_instrument_match.collect())
 
-        return metrics
+                    if isinstance(
+                        # pylint: disable=protected-access
+                        view_instrument_match._aggregation,
+                        _SumAggregation,
+                    ):
+                        data = Sum(
+                            aggregation_temporality=aggregation_temporality,
+                            data_points=view_instrument_match.collect(
+                                aggregation_temporality
+                            ),
+                            is_monotonic=isinstance(
+                                instrument, (Counter, ObservableCounter)
+                            ),
+                        )
+                    elif isinstance(
+                        # pylint: disable=protected-access
+                        view_instrument_match._aggregation,
+                        _LastValueAggregation,
+                    ):
+                        data = Gauge(
+                            data_points=view_instrument_match.collect(
+                                aggregation_temporality
+                            )
+                        )
+                    elif isinstance(
+                        # pylint: disable=protected-access
+                        view_instrument_match._aggregation,
+                        _ExplicitBucketHistogramAggregation,
+                    ):
+                        data = Histogram(
+                            data_points=view_instrument_match.collect(
+                                aggregation_temporality
+                            ),
+                            aggregation_temporality=aggregation_temporality,
+                        )
+                    elif isinstance(
+                        # pylint: disable=protected-access
+                        view_instrument_match._aggregation,
+                        _DropAggregation,
+                    ):
+                        continue
+
+                    metrics.append(
+                        Metric(
+                            # pylint: disable=protected-access
+                            name=view_instrument_match._name,
+                            description=view_instrument_match._description,
+                            unit=view_instrument_match._instrument.unit,
+                            data=data,
+                        )
+                    )
+                scope_metrics.append(
+                    ScopeMetrics(
+                        scope=instrument.instrumentation_scope,
```

Comment:
The ScopeMetrics need to be grouped together where the scope is the same, with their metrics being combined. The SDK should give fully grouped data out.